utilities/transactions/pessimistic_transaction_db.h

```diff
@@ -481,5 +492,38 @@ class WritePreparedTxnReadCallback : public ReadCallback {
   SequenceNumber snapshot_;
 };
 
+class WritePreparedTxnDBIterator final : public Iterator {
```

Comment:
The purpose of this class is not clear. It seems that ArenaWrappedDBIter would do the same. Where does it make use of snapshot_ and callback_?

Reply:
This wrapper iterator owns the snapshot and read callback and destroy them on its own destruction.

RAII is just a fancy acronym of: https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization

Reply:
1. How does it destroy the callback?
2. Can we add more comments to clarify the purpose of this class?

Reply:
1. By holding a unique_ptr of the callback.
2. will do.